test/TestHelpers.jl

```diff
@@ -95,7 +95,7 @@ function Base.similar(A::AbstractArray, T::Type, inds::Tuple{UnitRange,Vararg{Un
     OffsetArray(B, map(indsoffset, inds))
 end
 
-Base.similar(f::Union{Function,DataType}, shape::Tuple{UnitRange,Vararg{UnitRange}}) = OffsetArray(f(map(length, shape)), map(indsoffset, shape))
+Base.similar(f::Union{Function,Type}, shape::Tuple{UnitRange,Vararg{UnitRange}}) = OffsetArray(f(map(length, shape)), map(indsoffset, shape))
```

Comment:
should be `f::Base.Callable`? (why haven't we deprecated that yet?)

Since constructors are now attached to `UnionAll` not necessarily `DataType`, we will need to update the definition of `Callable` and the corresponding `notcalled_func` specialization test in gf.c